code/modules/integrated_electronics/subtypes/power.dm

```diff
@@ -41,7 +41,9 @@
 	if(!AM)
 		return FALSE
 	if(istype(AM, /obj/item/gun/energy))
-		return FALSE
+		var/obj/item/gun/energy/E = AM
+		if(!istype(AM.loc,/obj/item/integrated_circuit/manipulation/weapon_firing) || !E.can_charge)  // Can only charge guns that are inside of the assembly's weapon firing mechanism and chargeable
```

Comment:
they should not charge things in other assemblies. is there a check for that?

Reply:
There isn't, but there will be